File: frontend/packages/domain/lib/src/enums/game_status.dart
enum GameStatus {
  scheduled,
  inProgress,
  finished,
  cancelled;

  static GameStatus fromJson(String value) => switch (value) {
        'SCHEDULED' => GameStatus.scheduled,
        'IN_PROGRESS' => GameStatus.inProgress,
        'FINISHED' => GameStatus.finished,
        'CANCELLED' => GameStatus.cancelled,
        _ => throw FormatException('Status desconhecido: $value'),
      };

  String toJson() => switch (this) {
        GameStatus.scheduled => 'SCHEDULED',
        GameStatus.inProgress => 'IN_PROGRESS',
        GameStatus.finished => 'FINISHED',
        GameStatus.cancelled => 'CANCELLED',
      };

  /// Rótulo amigável em pt-BR para exibição na interface.
  String get label => switch (this) {
        GameStatus.scheduled => 'Agendado',
        GameStatus.inProgress => 'Ao vivo',
        GameStatus.finished => 'Encerrado',
        GameStatus.cancelled => 'Cancelado',
      };
}

File: backend/src/main/java/br/com/flagplatform/common/enums/GameStatus.java
package br.com.flagplatform.common.enums;

import lombok.Getter;

@Getter
public enum GameStatus implements PersistableEnum {

    SCHEDULED("SCHEDULED", "Scheduled"),
    IN_PROGRESS("IN_PROGRESS", "In progress"),
    FINISHED("FINISHED", "Finished"),
    CANCELLED("CANCELLED", "Cancelled");

    private final String code;
    private final String description;

    GameStatus(String code, String description) {
        this.code = code;
        this.description = description;
    }
}
